h/util/uri.py

```diff
@@ -167,15 +167,12 @@ def normalize(uristr):
     if uri.hostname is None:
         return uristr
 
-    scheme = _normalize_scheme(uri)
     netloc = _normalize_netloc(uri)
     path = _normalize_path(uri)
     query = _normalize_query(uri)
     fragment = None
 
-    uri = SplitResult(scheme, netloc, path, query, fragment)
-
-    return uri.geturl()
+    return SplitResult("httpx", netloc, path, query, fragment).geturl()
```

Comment:
If you look in the code above here you see that if `scheme not in URL_SCHEMES` we've already returned. So the test we make for `scheme in URL_SCHEMES` in `_normalize_scheme()` is redundant. It's not obvious, but we have unreachable code in there.